source/core-util_mbed_os.c

```diff
@@ -48,5 +48,30 @@ void core_util_assert_internal(const char *expr, const char *file, int line, con
     mbed_die();
 }
 
-#endif // #ifdef TARGET_LIKE_MBED
+#elif defined(TARGET_LIKE_POSIX)
```

Comment:
The idea behind the `core-util_mbed_os.c` file name was that it was implementing the required `core-util` functions only for mbed OS platforms. For POSIX, I'd create a new `core-util_posix.c` file.